Renaming Use, Migrate, MST and troubleshooting (#4193)
timescale/docs: mst/vpc-peering/vpc-peering-azure.md

# Configure VPC peering on Azure

You can Configure VPC peering for your $MST_LONG project,
using the VPC on Azure.

## Before you begin

*   Installed [Aiven Client][aiven-client-install].

> *   Signed in to your $MST_LONG dashboard.

Reviewer: ```suggestion
*   Signed in to $MST_CONSOLE_LONG.
```

    allow create other peerings later without assigning the role again for each
    VNet separately.

1.  Create a service principal for the $MST_LONG
    application object

    The $MST_LONG AD tenant contains an application object

>     similar to the one you created and $MST_LONG platform uses to

Reviewer: ```suggestion
    similar to the one you created, and $MST_LONG uses it to
```

>     create a peering from the Project VPC VNet in the $MST_LONG subscription to the

Reviewer: ```suggestion
    create a peering from the Project VPC VNet in $MST_LONG to the
```

>     VNet in your Azure subscription. For this the $MST_LONG app object needs a

Reviewer: ```suggestion
    VNet in Azure. For this, the $MST_LONG app object needs a
```

    include. Save the id  field from the output - this is referred to as
    `$aiven_role_id`.

> 1.  Assign the custom role to the $MST_LONG service principal to peer with your

Reviewer: ```suggestion
1.  Assign the custom role to the $MST_SERVICE_LONG principal to peer with your
```

>     VNet, assign the role that you created in the previous step to the $MST_LONG

Reviewer: ```suggestion
    VNet. Assign the role that you created in the previous step to the $MST_LONG
```

   Make note of the `tenantId` field from the output. It is referred to as `$user_tenant_id`.

1.  Create a peering connection from the $MST_LONG Project VPC using Aiven CLI:

    ```bash
    avn vpc peering-connection create --project-vpc-id $aiven_project_vpc_id --peer-cloud-account $user_subscription_id --peer-resource-group $user_resource_group --peer-vpc $user_vnet_name --peer-azure-app-id $user_app_id --peer-azure-tenant-id $user_tenant_id
    ```

    `$aiven_project_vpc_id` is the ID of the $MST_LONG project VPC, and can be
    found using the `avn vpc list` command.

>     The $MST_LONG platform creates a peering from the VNet in the $MST_LONG

Reviewer: ```suggestion
   $MST_LONG creates a peering from the VNet in the $MST_LONG
```

    Project VPC to the VNet in your subscription. In addition, it creates a
    service principal for the application object in your tenant
    `--peer-azure-app-id $user_app_id`, giving it permission to target the
    $MST_LONG subscription VNet with a peering. Your AD tenant ID is also needed
    in order for the $MST_LONG application object to authenticate with your
    tenant to give it access to the service principal that you created
    `--peer-azure-tenant-id $user_tenant_id`.

    Ensure that the arguments starting with `$user_` are in lower case. Azure
    resource names are case-agnostic, but the Aiven API currently only accepts
    names in lower case. If no error is shown, the peering connection is being set

>     up by the $MST_LONG platform.

Reviewer: ```suggestion
    up by $MST_LONG.
```

    ```

    A state such as `INVALID_SPECIFICATION`  or `REJECTED_BY_PEER`  may be shown
    if the VNet specified did not exist, or the $MST_LONG app object wasn't
    given permissions to peer with it. If that occurs, check your configuration
    and then recreate the peering connection. If everything went as expected,
    the state changes to `PENDING_PEER`  within a couple of minutes showing
    details to set up the peering connection from your VNet to the Project VPC's

>     VNet in the $MST_LONG subscription.

Reviewer: ```suggestion
    VNet in $MST_LONG.
```

    Now your application object has a session with both AD tenants

1.  Create a peering from your VNet to the VNet in the $MST_LONG subscription:

    ```bash
    az network vnet peering create --name <peering name of your choosing> --remote-vnet $aiven_vnet_id --vnet-name $user_vnet_name --resource-group $user_resource_group --subscription $user_subscription_id --allow-vnet-access
    ```

    If you do not specify `--allow-vnet-access` no traffic is allowed to flow

>     from the peered VNet and $MST_LONG services cannot be reached through the

Reviewer: ```suggestion
    from the peered VNet and $MST_SERVICE_LONGs cannot be reached through the
```

    avn vpc peering-connection get -v --project-vpc-id $aiven_project_vpc_id --peer-cl
    ```

>     The $MST_LONG platform polls peering connections in state `PENDING_PEER`

Reviewer: ```suggestion
   $MST_LONG polls peering connections in state `PENDING_PEER`
```
